presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -526,6 +527,69 @@ private static SqlScalarFunction decimalRoundNFunction()
                 .build();
     }
 
+    private static SqlScalarFunction decimalTruncateFunction()
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .name("truncate")
+                .literalParameters("num_precision", "num_scale", "result_precision")
+                .longVariableConstraints(
+                        longVariableCalculation("result_precision", "min(38, num_precision)"))
+                .argumentTypes("decimal(num_precision, num_scale)", BIGINT)
+                .returnType("decimal(result_precision, num_scale)")
```

Comment:
Precision should never increase.
We can reduce output scale to be ``Max(0, n)`` and precision to be ``num_prec - Max(0, n)``, but I thought it's not worth changing the number type in "rounding" function. Do you wan't me to do that?